### PR TITLE
Fix for comparisons with PackedArrays

### LINKED ISSUE #71073: Godot4 Array find() method works incorrectly with PackedVector2Arrays
### Godot version

v4.0.beta10.official [d0398f62f]

### System information

OS independent

### Issue description

It doesn't care about vectors in PackedVector2Array, it counts only array size.
Code example below is self-explaining. Any PackedVector2Array may be given as argument to Array.find() method, it will return an index of any PackedVector2Array with same PackedVector2Array.size()

### Steps to reproduce

Minimal reproduction example of code:
```
	var testarray := []
	testarray.append(PackedVector2Array([Vector2(10,10), Vector2(10,20), Vector2(20,20), Vector2(20,10)]))
	print(testarray.find(PackedVector2Array([Vector2(1,1), Vector2(1,2), Vector2(2,2), Vector2(2,1)])))
#outputs 0 (expected output -1)

```

```
	var testarray2 := []
	testarray2.append(PackedVector2Array([Vector2(10,20), Vector2(20,20), Vector2(20,10)]))
	testarray2.append(PackedVector2Array([Vector2(10,10), Vector2(10,20), Vector2(20,20), Vector2(20,10)]))
	print(testarray2.find(PackedVector2Array([Vector2(100,1), Vector2(1,200), Vector2(2,2), Vector2(2,111)])))
#outputs 1 (expected output -1)
```

### Minimal reproduction project

[array find test.zip](https://github.com/godotengine/godot/files/10368857/array.find.test.zip)

### PR DESCRIPTION
**Pull request by AThousandShips**
Fixed errors with parenthesis that breaks hash_compare comparison.

Edit: Went with the more future proof version not relying on putting parenthesis around the macro

This should be possible to directly apply to 3.x as the code is identical

3.x version #71103 

Fixes #71073

### CODE REVIEW

**Comment by AThousandShips** [comment]
Should probably add some test cases to make it sure, will take a look at that soon

**Comment by AThousandShips** [comment]
Cleaned up some more parenthesis, did some basic testing and I think nothing has broken, will see tomorrow if I can work out some good unit tests to ensure correctness

**Review by kleonc — APPROVED** [review]

**Comment by AThousandShips** [comment]
Got a 3.x version (currently waiting for CI on that one) but I'm not sure of the procedure for backporting changes

**Comment by akien-mga** [comment]
If the PR cherry-picks cleanly with `git cherry-pick`, we can just add the `cherrypick:3.x` label and one of the 3.x release managers can do it when doing a batch of cherry-picks.

If it needs to be redone manually because it's not trivial to cherry-pick, then you can open a PR directly, with the same commit message (you can add `[3.x]` in the PR title), and link both PRs in their OP for clarity.

**Comment by AThousandShips** [comment]
I don't think I'll be making any tests right now, busy at present, so go ahead and merge as is

**Comment by akien-mga** [comment]
Thanks!

**Comment by AThousandShips** [comment]
Thank you!

**Pull request merged**
